htdocs/ticket/class/ticket.class.php

```diff
@@ -570,7 +571,18 @@ public function create($user, $notrigger = 0)
 
 			if (!$error && !empty($conf->global->TICKET_ADD_AUTHOR_AS_CONTACT)) {
 				// add creator as contributor
-				if ($this->add_contact($user->id, 'CONTRIBUTOR', 'internal') < 0) {
+
+				// We first check the type of contact (internal or external)
+				if (!empty($user->socid) && !empty($user->contact_id) && $conf->global->TICKET_ADD_AUTHOR_AS_CONTACT == 2) {
```

Comment:
Can you replace $conf->global->TICKET_ADD_AUTHOR_AS_CONTACT
with
getDolGlobalInt('TICKET_ADD_AUTHOR_AS_CONTACT')

Reply:
I'll do but I also have to fix some things here so I'll do it for the next commit.